src/library.cpp

```diff
@@ -1103,7 +1103,7 @@ Library read_gds(const char* filename, double unit, double tolerance, const Set<
             case GdsiiRecord::XY:
                 if (polygon) {
                     polygon->point_array.ensure_slots(data_length / 2);
-                    double* d = (double*)polygon->point_array.items + polygon->point_array.count;
+                    double* d = (double*)polygon->point_array.items + 2 * polygon->point_array.count;
```

Comment:
Thanks for catching this! I'd prefer to write it like this, so that the intention (and my mistake) become clearer:
`double* d = (double*)&polygon->point_array.items[polygon->point_array.count];`
or
`double* d = (double*)(polygon->point_array.items + polygon->point_array.count);`

Reply:
Updated! Thank you for the quick response!